### DISCUSSION TITLE
Some questions on the ComputeFiniteStrain class

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#27196] [Q&A Modules: Solid mechanics]
Hi, I want to implement a custom material model. The problem to be solved is a Finite Strain problem, so I need to calculate the strain using ComputeFiniteStrain.

-Assuming my problem were Linear Elastic, I would use ComputeFiniteStrainElasticStress, which I understand computes the Cauchy Stress using an incremental formulation, by looking at the source code. I don't see where the properties needed are declared in ComputeFiniteStrain, i.e `elastic_strain` and `strain_increment `. How/where are they gotten from?

-My custom model is a hyperelastic material model, and the strain energy is calculated based on the principal stretches, lambda_(1,2,3). Therefore I will need the stretch tensor, U. I see it referred to in the documentation, but how do I "get" it? I cannot find it declared in ComputeFiniteStrain. 

Also, my problem works with the Green-Lagrange Strain, and second PK stress. To calculate the Green-Lagrange Strain I need the Cauchy-Green deformation tensor, C, which I can calculate from U.

-Are the names of the variables also the same for ComputeIncrementalSmallStrain?

-I guess this last one is more of a mechanics problem but by my understanding, for an elastic/hyperelastic problem the old strain + the strain increment are always equal to the current strain, right?

**Comment by GiudGiud** [comment]
@bwspenc

**Comment by jiangwen84** [comment]
> -Assuming my problem were Linear Elastic, I would use ComputeFiniteStrainElasticStress, which I understand computes the Cauchy Stress using an incremental formulation, by looking at the source code. I don't see where the properties needed are declared in ComputeFiniteStrain, i.e `elastic_strain` and `strain_increment `. How/where are they gotten from?
> 
Look at it base class, i.e. ComputeIncrementalStrainBase.C

> -My custom model is a hyperelastic material model, and the strain energy is calculated based on the principal stretches, lambda_(1,2,3). Therefore I will need the stretch tensor, U. I see it referred to in the documentation, but how do I "get" it? I cannot find it declared in ComputeFiniteStrain.
>  
Same, look for the base class, if not available, you can declare and compute it in your new material class. 
> Also, my problem works with the Green-Lagrange Strain, and second PK stress. To calculate the Green-Lagrange Strain I need the Cauchy-Green deformation tensor, C, which I can calculate from U.
> 

Same. 
> -Are the names of the variables also the same for ComputeIncrementalSmallStrain?
>

 What are the variable names you refer to?
> -I guess this last one is more of a mechanics problem but by my understanding, for an elastic/hyperelastic problem the old strain + the strain increment are always equal to the current strain, right?

In In principle , Yes.
